src/asset.ts

```diff
@@ -66,54 +29,108 @@ export default class Asset extends AinftBase {
     contractAddress?: string,
     tokenId?: string,
   ) {
-    const timestamp = Date.now();
-    const query: any = { appId };
-    if (contractAddress) query.contractAddress = contractAddress;
-    if (tokenId) query.tokenId = tokenId;
+    const query: any = {
+      appId,
+      ...contractAddress && { contractAddress },
+      ...tokenId && { tokenId },
+    };
     const trailingUrl = `nft/${chainId}/${ethAddress}`;
-    const data = buildData(
-      HttpMethod.GET,
-      `/asset/${trailingUrl}`,
-      timestamp,
-      query
-    );
-    const signature = this.signData(data);
-    return axios
-      .get(`${this.baseUrl}/${trailingUrl}`, {
-        params: query,
-        headers: {
-          'X-AINFT-Date': timestamp,
-          Authorization: `AINFT ${signature}`,
-        },
-      })
-      .then((res) => res.data.data)
-      .catch((e) => {
-        throw e.response.data;
-      });
+    const userNfts = (await this.sendRequest(HttpMethod.GET, prefix, trailingUrl, query)) as UserNfts;
```

Comment:
could we just return without assigning?

Reply:
I tried to specify the return type for the convenience of the user, but it would be better to just specify it when declaring the function.

Reply:
Fixed, thanks!